### PR TITLE
Add test coverage for checkbox

### PR DESCRIPTION
**Pull request by Eudora-Li01**
Related #https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for CheckBox to test its events: AppearanceChanged, DoubleClick, MouseDoubleClick, CheckedChanged and CheckedStateChanged.

- Add AccessibleObject unit test for CheckBox to test its methods: DefaultAction.
- Update AccessibleObject unit test for CheckBox to test its methods: State.

## Customer Impact
No

## Regression? 

- No

## Risk

Minimal

## Test methodology 

Unit Tests

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11249)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CheckBoxTests.cs [hunk 1]
```diff
@@ -492,8 +695,13 @@ public new bool ResizeRedraw
 
         public new bool GetTopLevel() => base.GetTopLevel();
 
+        public new void OnAppearanceChanged(EventArgs e) => base.OnAppearanceChanged(e);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CheckBoxTests.cs:698]: Maybe me we can add blank line between these methods.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CheckBoxTests.cs [hunk 2]
```diff
@@ -492,8 +695,13 @@ public new bool ResizeRedraw
...
         public new void OnClick(EventArgs e) => base.OnClick(e);
+        public new void OnDoubleClick(EventArgs e) => base.OnDoubleClick(e);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/CheckBoxTests.cs:700]: same as above, and also applied for below rest methods.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs [hunk 3]
```diff
@@ -18,8 +18,8 @@ public void CheckBoxAccessibleObject_Ctor_NullControl_ThrowsArgumentException()
     [WinFormsFact]
     public void CheckBoxAccessibleObject_Ctor_InvalidTypeControl_ThrowsArgumentException()
     {
-        using TextBox textBox = new();
-        Assert.Throws<ArgumentException>(() => new CheckBoxAccessibleObject(textBox));
+        using CheckBox checkBox = new();
+        Assert.Throws<ArgumentException>(() => new CheckBoxAccessibleObject(checkBox));
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs:22]: Moving towards https://fluentassertions.com/ in the newly added tests


For a detailed guide, refer to the Fluent Assertions documentation [here](https://fluentassertions.com/introduction).

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs [hunk 4]
```diff
@@ -34,19 +34,41 @@ public void CheckBoxAccessibleObject_Ctor_Default()
     }
 
     [WinFormsFact]
-    public void CheckBoxAccessibleObject_CustomDoDefaultAction_ReturnsExpected()
+    public void CheckBoxAccessibleObject_DefaultAction_IfHandleIsNotCreated_ReturnsNull()
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs:37]: The verification content does not match the test title

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs [hunk 5]
```diff
@@ -34,19 +34,41 @@ public void CheckBoxAccessibleObject_Ctor_Default()
...
         CheckBoxAccessibleObject checkBoxAccessibleObject = new(checkBox);
 
-        Assert.Equal("TestActionDescription", checkBoxAccessibleObject.DefaultAction);
-        Assert.False(checkBox.IsHandleCreated);
+        checkBox.AccessibilityObject.DefaultAction.Should().Be("TestActionDescription");
+        checkBox.IsHandleCreated.Should().BeFalse();
+    }
+
+    [WinFormsFact]
+    public void CheckBoxAccessibleObject_DefaultAction_IfHandleIsCreatedAndCheckedIsTrue_ReturnsExpected()
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs:62]: This test can be combined with `CheckBoxAccessibleObject_DefaultAction_IfHandleIsNotCreated_ReturnsNull()`

Comment (Eudora-Li01) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/AccessibleObjects/CheckBox.CheckBoxAccessibleObjectTests.cs]: Combined those two methods and refactor method `CheckBoxAccessibleObject_DefaultAction_CheckedIsTrue_ReturnsExpected`

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**